gcore/gdaldriver.cpp

```diff
@@ -949,16 +960,56 @@ GDALDataset *GDALDriver::CreateCopy( const char * pszFilename,
     if( pfnProgress == nullptr )
         pfnProgress = GDALDummyProgress;
 
+/* -------------------------------------------------------------------- */
+/*      If no INTERLEAVE creation option is given, we will try to add   */
+/*      one that matches the current srcDS interleaving                 */
+/* -------------------------------------------------------------------- */
+    char** papszOptionsSmart = 0;
```

Comment:
to modify options I would suggest you use the existing pattern of char** papszOptionsToDelete = nullptr;   

char** papszOptionsToDelete = nullptr;   
if (we need to modify options )
{
    papszOptionsToDelete = CSLDuplicate(papszOptions);
    // alter papszOptionsToDelete with CSLSetNameValue()
    papszOptions = papszOptionsToDelete;
}

Note: you'll need to add a "if ( papszOptionsToDelete == nullptr )" before "papszOptionsToDelete = CSLDuplicate(papszOptions);" in the branch if( iIdxQuietDeleteOnCreateCopy >= 0 )   (it was the first one to alter the options, so didn't need to test that)